vllm/entrypoints/openai/serving_engine.py

```diff
@@ -153,6 +168,25 @@ def _maybe_get_adapters(
         for prompt_adapter in self.models.prompt_adapter_requests:
             if request.model == prompt_adapter.prompt_adapter_name:
                 return None, prompt_adapter
+        if request.model and self.lora_cache_dir is not None:
+            adapter_path = os.path.join(self.lora_cache_dir, request.model)
+            adapter_config_path = os.path.join(adapter_path,
+                                               "adapter_config.json")
+            if os.path.exists(adapter_config_path):
+                with open(adapter_config_path) as file:
+                    adapter_config = json.load(file)
+                if adapter_config["peft_type"] == "LORA":
+                    lora_request = LoadLoRAAdapterRequest(
+                        lora_name=request.model,
+                        lora_path=adapter_path,
+                    )
+                    response = await self.models.load_lora_adapter(lora_request
+                                                                   )
+                    if isinstance(response, ErrorResponse):
+                        raise ValueError(response.message)
```

Comment:
The error returned if an adapter fails to load looks like:
```
{
  "object": "error",
  "message": "The size of tensor a (2560) must match the size of tensor b (4096) at non-singleton dimension 1",
  "type": "BadRequestError",
  "param": null,
  "code": 400
}
```

We could add a prefix to the message to provide a little more context:
```
 ValueError(f"Failed to load LoRA adapter: {response.message}")
```